storage/innobase/row/row0merge.cc

```diff
@@ -4718,6 +4718,9 @@ row_merge_build_indexes(
 				error = btr_bulk.finish(error);
 			}
 		}
+		if (error == DB_SUCCESS) {
+			row_merge_write_redo(indexes[i]);
+		}
```

Comment:
it is wrong to write redo before flush.

Also, is it possible to reach here without the btr bluk ? How do we truly know that we have written rows using btr bluk insert?

Reply:
Now doing it inside the if block. IIUC it would be using btr bulk  except when building FTS index.

Reply:
@rahulmalik87 the patch is still wrong. Writing redo log record before flush.

Reply:
Also better to do it in one place instead of duplicating in all branches. Just check if btrbulk is inited and records written using btr bulk

Reply:
removed the duplicating in all branch